wgpu-core/src/pipeline.rs

```diff
@@ -183,13 +183,27 @@ pub enum CreateRenderPipelineError {
     },
     #[error("missing required device features {0:?}")]
     MissingFeature(wgt::Features),
-    #[error("error in stage {flag:?}: {error}")]
+    #[error("error in stage {flag:?}")]
     Stage {
         flag: wgt::ShaderStage,
+        #[source]
         error: StageError,
     },
 }
 
+impl CreateRenderPipelineError {
+    pub(crate) fn context<'a>(
```

Comment:
Can we iterate a bit on this name? Generally, no-verb methods should be used for pure accessor functions.
Maybe this one should be `with_label`?

Reply:
I'm fine with anything, I originally called it `named` but that felt a bit disconnected what it was doing. Then I thought I was adding more context to the error, akin to the `context` in anyhow.